src/TreeSitter/Node.hs

```diff
@@ -29,10 +27,9 @@ data Node = Node
 data TSPoint = TSPoint { pointRow :: !Word32, pointColumn :: !Word32 }
   deriving (Show, Eq, Generic)
 
-data TSNode = TSNode !(Ptr ()) !Word32 !Word32
+data TSNode = TSNode !Word32 !Word32 !Word32 !Word32 !(Ptr ()) !(Ptr ())
```

Comment:
What are these fields? Is there overlap with the fields of `Node`? Could we 🔥 some of the `Node` fields in favour of some of these?

Reply:
These are new reserved data that `ts_node` uses for internal state. We should never touch them. Ideally we could omit them entirely, but then the Storable instance would not be lawful.